### DISCUSSION TITLE
copy and paste between remote maps

### DISCUSSION THREAD

**Discussion by avatarUNO** [freeplane/freeplane#1609] [Help]
Hello!. 

First of all, warn 3 points.

    1. my native language is not English.
    2. This is my first post on this forum, so I don't know if this question has been asked before.
    3. My knowledge of programming in general, and in particular as in Grovvy, is very basic.

Below, I present my request for help, which is reflected in the title of the post.

This is the structure of my main map:

- MainMap
    - script
    - Father
       - son1
       - **son2**
           - blablaba..123.. 
       - son3 

I want the first node, called "script", above the "Father" node. Make a copy of the child nodes of the node named "Father".
Then it sends them to their respective maps that I have in another folder on my PC.
For example:

  - freeplane:/%20/C:/Users/usuario/Documents/freeplane/son1.mm#ID_0000000000
  - freeplane:/%20/C:/Users/usuario/Documents/freeplane/son2.mm#ID_1111111111
  - freeplane:/%20/C:/Users/usuario/Documents/freeplane/son1.mm#ID_1111100000

PD: _Actually this is just the beginning of what I really want, but for starters I think it's enough_ 

Can someone make a simple script for me?

**Comment by EdoFro** [comment]
> Can someone make a simple script for me?

Hi @avatarUNO,
It's not a simple script what you are asking for. Mostly because there are a lot of definitions you have to make to limit the scripts functionality.

I think you can ask for help to solve some steps of the script, and then you can put all that together.

First questions:
maps son1.mm, son2.mm and son3.mm are also open in Freeplane when executing the script?
if not: should Freeplane open them? visible or hidden?

with `c.openMindMaps` you get a list of all opened maps in Freeplane

and with `c.openMindMaps*.name` a list of their names


this way you get an open map by its name:
```groovy
def targetMapName = 'Map2'
def targetMap = c.openMindMaps.find{it.name == targetMapName}
```

on the other hand, here is an example of how to copy a branch
```groovy
def nodo = node
def targetNode = nodo.parent //just an  example

targetNode.appendBranch(nodo)
```

one combining both features:

```groovy
def nodo = node //selected node in the active map

def targetMapName = 'Map2' // change this with your map's name
def targetMap = c.openMindMaps.find{it.name == targetMapName}

def targetNode = targetMap.root // in this case it uses the root node as target

targetNode.appendBranch(nodo)

ui.informationMessage('branch copy/pasted to target map')
```

hope this helps,

edo
